### PR TITLE
decouple builder from config

### PR DESCRIPTION
**Pull request by niktrix**
## Problem

As we are separting out P2P connection, netork builder should take address as value instead of specific config

## Testing Done and Results

updated test cases

### CODE REVIEW

File: p2p/network/builder.go [hunk 1]
```diff
@@ -124,12 +123,10 @@ func WriteTimeout(d time.Duration) BuilderOption {
 }
 
 // NewBuilder returns a new builder with default options.
-func NewBuilder(env string) *Builder {
-	config := config.GetConfiguration(env)
-	address := config.ConstructTCPAddress()
+func NewBuilder(fullTCPAddress string) *Builder {
```
Comment (cuonglm) [inline p2p/network/builder.go]: I don't thing passing address here is good, please make it empty (Also the name `fullTCPAddress` is something weird).

File: p2p/network/builder.go [hunk 2]
```diff
@@ -142,8 +139,8 @@ func NewBuilder(env string) *Builder {
 }
 
 // NewBuilderWithOptions returns a new builder with specified options.
-func NewBuilderWithOptions(env string, opt ...BuilderOption) *Builder {
-	builder := NewBuilder(env)
+func NewBuilderWithOptions(fullTCPAddress string, opt ...BuilderOption) *Builder {
```
Comment (cuonglm) [inline p2p/network/builder.go]: Please don't do this, make the address as a `BuilderOption` instead.

Comment (niktrix) [reply p2p/network/builder.go]: Adress is needed before creating builder, can you tell me how can I access address if it is in Builderoption

Comment (cuonglm) [reply p2p/network/builder.go]: @niktrix please do as our discussion on slack.

**Pull request merged**